sdgym/datasets.py

```diff
@@ -62,94 +95,161 @@ def _get_dataset_path(
+    for obj in contents:
+        key = obj.get('Key', '')
+        if key.endswith('metainfo.yaml'):
+            parts = key.split('/')
+            if len(parts) >= 3:
+                dataset_name = parts[-2]
+                yaml_key = key
+                info = _load_yaml_metainfo_from_s3(s3_client, bucket_name, yaml_key)
+                size_mb = _parse_numeric_value(
+                    info.get('dataset-size-mb', np.nan),
+                    dataset_name,
+                    field_name='dataset-size-mb',
+                    target_type=float,
+                )
+                num_tables = _parse_numeric_value(
+                    info.get('num-tables', np.nan), dataset_name, 'num-tables', target_type=int
+                )
+                tables_info['dataset_name'].append(dataset_name)
+                tables_info['size_MB'].append(size_mb)
+                tables_info['num_tables'].append(num_tables)
+
+    return tables_info
+
+
+def _get_available_datasets(
+    modality, bucket=None, aws_access_key_id=None, aws_secret_access_key=None
+):
+    _validate_modality(modality)
+    s3_client = get_s3_client(aws_access_key_id, aws_secret_access_key)
+    bucket = bucket or BUCKET
+    bucket_name = _get_bucket_name(bucket)
+    contents = _list_s3_bucket_contents(s3_client, bucket_name, f'{modality}/')
+    datasets_info = _genereate_dataset_info(s3_client, bucket_name, contents)
+    return pd.DataFrame(datasets_info)
+
+
 def load_dataset(
     modality,
     dataset,
     datasets_path=None,
     bucket=None,
     aws_access_key_id=None,
     aws_secret_access_key=None,
-    limit_dataset_size=None,
+    limit_dataset_size=False,
 ):
     """Get the data and metadata of a dataset.
 
     Args:
         modality (str):
-            It must be ``'single-table'``, ``'multi-table'`` or ``'time-series'``.
+            It must be ``'single_table'``, ``'multi_table'`` or ``'sequential'``.
         dataset (str):
-            The path of the dataset as a string.
+            The path of the dataset as a string or the name of the dataset.
         dataset_path (PurePath):
             The path of the dataset as an object. This will only be used if the given ``dataset``
             doesn't exist.
         bucket (str):
             The AWS bucket where to get the dataset. This will only be used if both ``dataset``
             and ``dataset_path`` don't exist.
-        aws_access_key_id (str):
+        aws_access_key_id (str or None):
             The access key id that will be used to communicate with s3, if provided.
-        aws_secret_access_key (str):
+        aws_secret_access_key (str or None):
             The secret access key that will be used to communicate with s3, if provided.
         limit_dataset_size (bool):
             Use this flag to limit the size of the datasets for faster evaluation. If ``True``,
             limit the size of every table to 1,000 rows (randomly sampled) and the first 10
```

Comment:
It says the data is "randomly sample" when it's actually just its `head`.

Reply:
Done in https://github.com/sdv-dev/SDGym/pull/471/commits/6df617b2c703224a7a4437930e189028d6763c4d